component/core/src/test/java/org/exoplatform/social/core/listeners/FileActionListenerTest.java

```diff
@@ -0,0 +1,92 @@
+package org.exoplatform.social.core.listeners;
+
+import org.exoplatform.commons.file.model.FileInfo;
+import org.exoplatform.commons.file.model.FileItem;
+import org.exoplatform.commons.file.services.FileService;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.test.AbstractCoreTest;
+import org.exoplatform.social.metadata.MetadataService;
+import org.exoplatform.social.metadata.model.MetadataItem;
+import org.exoplatform.social.metadata.model.MetadataKey;
+import org.exoplatform.social.metadata.model.MetadataType;
+import org.exoplatform.social.metadata.thumbnail.model.ThumbnailObject;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FileActionListenerTest extends AbstractCoreTest {
+
+  private MetadataService    metadataService;
+
+  private IdentityManager    identityManager;
+
+  private FileService        fileService;
+
+  private Identity           userIdentity;
+
+  private FileActionListener fileActionListener;
+
+  public void setUp() throws Exception {
+    super.setUp();
+    metadataService = getContainer().getComponentInstanceOfType(MetadataService.class);
+    identityManager = getContainer().getComponentInstanceOfType(IdentityManager.class);
+    userIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, "john");
+    fileService = getContainer().getComponentInstanceOfType(FileService.class);
+
+    fileActionListener = new FileActionListener(metadataService);
+  }
+
+  public void testOnEvent() throws Exception {
+    ThumbnailObject thumbnailObject = new ThumbnailObject("file", "1");
```

Comment:
Your test here is not a unit test, it is integration test.
You are verifying that, when creating the file by file service, the event is correctly launched. So your are testing the ListenerService.

If you want to unit test the FileActionServiceListener, you should do something like that :
FileActionListener f = new fileActionListener();

Event<FileInfo, Object> event = new Event ....
event.fileInfo.setEventName=FILE_UPDATED
f.onEvent(event);
check that deleteMetadataItemsByMetadataTypeAndObject was called

same with eventName = FILE_DELETED